test/server/cards/01_SOR/LukeSkywalkerFaithfulFriend.spec.js

```diff
@@ -0,0 +1,99 @@
+describe('Luke Skywalker, Faithful Friend', function() {
+    integration(function() {
+        describe('Luke\'s undeployed ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['battlefield-marine', 'cartel-spacer', 'alliance-xwing'],
+                        groundArena: ['atst'],
+                        leader: 'luke-skywalker#faithful-friend'
+                    },
+                    player2: {
+                        hand: ['alliance-dispatcher'],
+                        groundArena: ['specforce-soldier'],
+                    }
+                });
+            });
+
+            it('should give a friendly heroism unit played by us this turn a shield token', function () {
+                this.player1.clickCard(this.battlefieldMarine);
+
+                this.player2.clickCard(this.allianceDispatcher);
+
+                this.player1.clickCard(this.cartelSpacer);
+
+                this.player2.passAction();
+
+                const resourcesSpentBeforeLukeActivation = this.player1.countExhaustedResources();
+                this.player1.clickCard(this.lukeSkywalker);
+                this.player1.clickPrompt('Give a shield to a heroism unit you played this phase');
+                expect(this.battlefieldMarine).toHaveExactUpgradeNames(['shield']);
+                expect(this.lukeSkywalker.exhausted).toBe(true);
+                expect(this.player1.countExhaustedResources()).toBe(resourcesSpentBeforeLukeActivation + 1);
```

Comment:
Is this the only test covering a leader cost? I assume this would be covered elsewhere and doesn't really need to be tested here?

Reply:
I actually consider that cost part of the ability since it's something that has to explicitly written in the implementation file. Since part of the point of unit tests is to make sure that the card writer didn't have some typo and forget something, I'd always want someone writing a test with a resource cost to validate that the cost is paid (same as checking that Luke is exhausted since that's the other part of the cost). Does that make sense?